src/calling/haplotypes.rs

```diff
@@ -225,62 +210,67 @@ impl HaplotypeVariants {
                 let mut s = String::new();
                 x.read_to_string(&mut s);
                 if haplotypes.contains(&s) {
-                    haplotype_indices.push(sample_index);
+                    haplotype_indices.push(sample_index); //pushing the exact indices of haplotypes that are found in header.samples()
                 }
             }
+
             let mut bv: BitVec<usize> = BitVec::new();
 
             for sample_index in haplotype_indices.iter() {
                 for gta in gts.get(*sample_index).iter() {
-                    if gta == &Unphased(1) {
-                        bv.push(true);
-                    } else {
-                        bv.push(false);
-                        break;
-                    }
+                    bv.push(*gta == Unphased(1));
                 }
             }
-            candidate_variants.insert(Variant(variant.clone()), bv);
+            candidate_variants.insert(VariantID(variant_id.clone()), bv);
         }
         Ok(HaplotypeVariants(candidate_variants))
     }
 }
+#[derive(Debug, Clone, Derefable)]
+pub(crate) struct AlleleFreqDist(#[deref] BTreeMap<AlleleFreq, f64>);
+
+impl AlleleFreqDist {
+    pub(crate) fn vaf_query(&self, vaf: AlleleFreq, likelihood: &Prob) -> Result<Prob> {
+        if self.contains_key(&vaf) {
+            let likelihood = likelihood * Prob::from(PHREDProb(*self.get(&vaf).unwrap()));
+        } else {
+            let (x_0, y_0) = self.range(..vaf).next_back().unwrap();
+            let (x_1, y_1) = self.range(vaf..).next().unwrap();
+
+            let density = NotNan::new(*y_0).unwrap() + (vaf - *x_0) * (*y_1 - *y_0) / (*x_1 - *x_0); //calculation of density for given vaf by linear interpolation
+            let likelihood = likelihood * Prob::from(PHREDProb(NotNan::into_inner(density)));
+        }
+        Ok(*likelihood)
```

Comment:
Here you just return the likelihood that was given as an argument. But instead, there should be no such argument and you should return what you have computed in the if-else.